posthog/api/external_web_analytics/query_adapter.py

```diff
@@ -0,0 +1,117 @@
+from datetime import date, datetime
+from typing import Optional, Any
+
+from posthog.hogql_queries.web_analytics.web_overview import WebOverviewQueryRunner
+from posthog.schema import (
+    PersonPropertyFilter,
+    SessionPropertyFilter,
+    WebOverviewQuery,
+    DateRange,
+    HogQLQueryModifiers,
+    EventPropertyFilter,
+    PropertyOperator,
+    WebOverviewQueryResponse,
+)
+from posthog.models import Team
+from posthog.api.external_web_analytics.serializers import WebAnalyticsOverviewRequestSerializer
+
+
+class ExternalWebAnalyticsQueryAdapter:
+    """
+    Adapter that uses the internal WebOverviewQueryRunner to provide data for the external API.
+    It tries to separate the web analytics query runners from the external API.
+    """
+
+    def __init__(self, team: Team):
+        self.team = team
+
+    def _get_base_properties(
+        self, domain: Optional[str] = None
+    ) -> list[EventPropertyFilter | PersonPropertyFilter | SessionPropertyFilter]:
+        properties = []
+        if domain:
+            properties.append(
+                EventPropertyFilter(
+                    key="$host",
+                    operator=PropertyOperator.EXACT,
+                    value=[domain],
+                )
+            )
+        return properties
```

Comment:
I suppose the bot is right, but I would assume being more flexible with the type union would not be a problem. 

That said, we don't support PersonProperties and just some session properties on the pre-aggregated tables, here is the current mapping that we do: https://github.com/PostHog/posthog/blob/a64d3ffb9b083dbac7440f2fa5b776a747715f63/posthog/hogql_queries/web_analytics/pre_aggregated/properties.py#L1-L74

Since we said that we would want those endpoints to be more strict, I think we should default to only allowing those instead of the whole property taxonomy. WDYT?